src/ethereum_test_tools/common/types.py

```diff
@@ -785,6 +872,316 @@ def with_fields(self, **kwargs) -> "Transaction":
                 raise ValueError(f"Invalid field '{key}' for Transaction")
         return tx
 
+    def payload_body(self) -> List[Any]:
+        """
+        Returns the list of values included in the transaction body.
+        """
+        if self.signature is None:
+            raise ValueError("signature must be set before serializing any tx type")
+
+        if self.gas_limit is None:
+            raise ValueError("gas_limit must be set for all tx types")
+        to = address_to_bytes(self.to)
+
+        if self.ty == 3:
+            # EIP-4844: https://eips.ethereum.org/EIPS/eip-4844
+            if self.max_priority_fee_per_gas is None:
+                raise ValueError("max_priority_fee_per_gas must be set for type 3 tx")
+            if self.max_fee_per_gas is None:
+                raise ValueError("max_fee_per_gas must be set for type 3 tx")
+            if self.max_fee_per_data_gas is None:
+                raise ValueError("max_fee_per_data_gas must be set for type 3 tx")
+            if self.blob_versioned_hashes is None:
+                raise ValueError("blob_versioned_hashes must be set for type 3 tx")
+
+            if self.network_version:
+                if self.blobs is None:
+                    raise ValueError("blobs must be set for network version of type 3 tx")
+                if self.blob_kzg_commitments is None:
+                    raise ValueError(
+                        "blob_kzg_commitments must be set for network version of type 3 tx"
+                    )
+                if self.blob_kzg_proofs is None:
+                    raise ValueError(
+                        "blob_kzg_proofs must be set for network version of type 3 tx"
+                    )
+
+                return [
+                    [
+                        Uint(self.chain_id),
+                        Uint(self.nonce),
+                        Uint(self.max_priority_fee_per_gas),
+                        Uint(self.max_fee_per_gas),
+                        Uint(self.gas_limit),
+                        to,
+                        Uint(self.value),
+                        code_to_bytes(self.data),
+                        [a.to_list() for a in self.access_list]
+                        if self.access_list is not None
+                        else [],
+                        Uint(self.max_fee_per_data_gas),
+                        [hash_to_bytes(h) for h in self.blob_versioned_hashes],
+                        Uint(self.signature[0]),
+                        Uint(self.signature[1]),
+                        Uint(self.signature[2]),
```

Comment:
We could create a Payload class that we use throughout, with optional fields depending on the transaction type! 

Very happy with how it is now though, its very read-able :)

Reply:
Definitely some room for improvement, yes I like the idea of separating this into its own object.

I think in the future this call should somehow be fork-aware, because some of the transaction types will be updated to SSZ after a given fork.